### PR TITLE
DevKit updates for 4.x branch

### PR DESCRIPTION
**Pull request by SonataCI**
## Changelog

```markdown
### Removed
- Support for PHP 7.4
- Support for Symfony 6.0 and 6.1
- Support for `doctrine/persistence` ^2.0.
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
2 errors missing from psalm, not sure how to fix them @VincentLanglet , can you help me?

**Comment by VincentLanglet** [comment]
> 2 errors missing from psalm, not sure how to fix them @VincentLanglet , can you help me?

Sure:
```
ERROR: InvalidArgument - src/CmsManager/CmsSnapshotManager.php:154:56 - Argument 1 of Sonata\BlockBundle\Util\RecursiveBlockIteratorIterator::__construct expects Doctrine\Common\Collections\Collection<array-key, mixed>|array<array-key, mixed>, but Doctrine\Common\Collections\Collection<array-key, Sonata\PageBundle\Model\PageBlockInterface> provided (see https://psalm.dev/004)
        $iterator = new RecursiveBlockIteratorIterator($page->getBlocks());
```
This error need to be fixed in BlockBundle by changing `Collection` phpdoc to `ReadableCollection` in order to accept covariance. (You can try with https://github.com/sonata-project/SonataBlockBundle/pull/1142)

And for
```
$template = new Template('page', 'template.twig', ['header' => [
            'name' => 'Header',
            'block' => ['text.block'],
        ]]);
```
we're passing `block` and not `blocks`.

File: tests/Model/TemplateTest.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function testGetContainer(): void
     {
         $template = new Template('page', 'template.twig', ['header' => [
             'name' => 'Header',
-            'block' => ['text.block'],
+            'blocks' => ['text.block'],
```
Comment (jordisala1991) [inline tests/Model/TemplateTest.php]:
```suggestion
```

File: composer.json [hunk 2]
```diff
@@ -21,10 +21,10 @@
     ],
     "homepage": "https://docs.sonata-project.org/projects/SonataPageBundle",
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
         "cocur/slugify": "^4.0",
-        "doctrine/doctrine-bundle": "^2.5",
-        "doctrine/persistence": "^2.1 || ^3.0",
+        "doctrine/doctrine-bundle": "^2.7",
+        "doctrine/persistence": "^3.0",
```
Comment (jordisala1991) [inline composer.json]:
```suggestion
        "doctrine/persistence": "^3.0.2",
```

File: composer.json [hunk 3]
```diff
@@ -63,18 +62,19 @@
...
         "symfony/panther": "^1.0 || ^2.0",
-        "symfony/phpunit-bridge": "^6.1",
-        "vimeo/psalm": "^4.7.2"
+        "symfony/phpunit-bridge": "^6.2",
+        "symfony/yaml": "^4.4 || ^5.4 || ^6.2",
+        "vimeo/psalm": "^4.7.2 || ^5.0"
     },
     "conflict": {
-        "doctrine/orm": "<2.10"
+        "doctrine/orm": "<2.14"
```
Comment (jordisala1991) [inline composer.json:78]:
```suggestion
        "doctrine/annotations": "<1.13.3",
        "doctrine/orm": "<2.14"
```

File: composer.json [hunk 4]
```diff
@@ -63,18 +62,20 @@
...
         "symfony/panther": "^1.0 || ^2.0",
-        "symfony/phpunit-bridge": "^6.1",
-        "vimeo/psalm": "^4.7.2"
+        "symfony/phpunit-bridge": "^6.2",
+        "symfony/yaml": "^4.4 || ^5.4 || ^6.2",
+        "vimeo/psalm": "^4.7.2 || ^5.0"
     },
     "conflict": {
-        "doctrine/orm": "<2.10"
+        "doctrine/annotations": "<1.13.3",
```
Comment (jordisala1991) [inline composer.json]:
```suggestion
        "doctrine/dbal": "<3.4",
```

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**